src/persist/src/persist.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// See https://developers.google.com/protocol-buffers for what's going on here.
+
+syntax = "proto3";
+
+package gen;
+
+message ProtoMeta {
+    uint64 seqno = 2;
+    repeated ProtoStreamRegistration id_mapping = 3;
+    repeated ProtoStreamRegistration graveyard = 4;
+    repeated ProtoArrangement arrangements = 5;
+
+    // TODO: For data written by some vX.Y.Z of Materialize, we'll support
+    // reading it back in by later versions (backward compatibility, so users
+    // can upgrade) and earlier versions (forward compatiblity, so users can
+    // roll back an upgrade). The specific policy is yet to be determined, but
+    // each of these is likely to be bounded, especially forward
+    // compatatibility.
+    //
+    // For us to reason about this (e.g. to prevent startup if a binary is
+    // pointed at data it can't handle), we'll want to store the version that
+    // wrote data alongside the data itself. Figure out whether this should be
+    // the version triple (and how to plumb that down to persist) or something
+    // else.
+    reserved 1;
+}
+
+message ProtoStreamRegistration {
+    uint64 id = 1;
+    string name = 2;
+    string key_codec_name = 3;
+    string val_codec_name = 4;
+}
+
+message ProtoArrangement {
+    uint64 id = 1;
+    ProtoU64Antichain since = 2;
+    ProtoU64Antichain seal = 3;
+    repeated ProtoUnsealedBatchMeta unsealed_batches = 4;
+    repeated ProtoTraceBatchMeta trace_batches = 5;
+}
+
+message ProtoUnsealedBatchMeta {
+    string key = 1;
```

Comment:
I remember us talking about how you wanted to store uuids directly instead of strings at some point. did you still want to do that / would that change the proto type used here?

Reply:
yup, i went to do it and it's a little annoying for Reasons (tm), so i held off. i was actually considering using it as test run for making a backward compatible change after this lands (and then cleaning up the cruft next time we wipe the data)

Reply:
that sounds fine to me